misc/dbt-materialize/tests/adapter/test_clusters.py

```diff
@@ -100,13 +101,25 @@ def test_materialize_override_ok(self, project):
         assert len(results) == 1
 
         project.run_sql("CREATE CLUSTER not_default REPLICAS (r1 (SIZE '1'))")
-        run_dbt(["run", "--exclude", "invalid_cluster"])
+        run_dbt(["run", "--exclude", "invalid_cluster", "default_cluster"])
 
         check_relations_equal(project.adapter, ["actual_clusters", "expected_clusters"])
 
     def test_materialize_override_noexist(self, project):
         run_dbt(["run", "--models", "invalid_cluster"], expect_pass=False)
 
+    # In the absence of the pre-installed `default` cluster, Materialize should
+    # not error if a user-provided cluster is specified as a connection or
+    # model config, but will error otherwise.
+    # See #17197: https://github.com/MaterializeInc/materialize/pull/17197
+    def test_materialize_drop_default(self, project):
+        project.run_sql("DROP CLUSTER default CASCADE")
+
+        run_dbt(["run", "--models", "override_cluster"], expect_pass=True)
+        run_dbt(["run", "--models", "default_cluster"], expect_pass=False)
+
+        project.run_sql("CREATE CLUSTER default REPLICAS (r1 (SIZE '1'))")
```

Comment:
There is a tiny risk of confusion here, in that if the test fails partway through, it won't recreate the `default` cluster, and then the following tests in the run will fail in a confusing way. Probably not worth fixing unless this test framework has an easy way of associating a cleanup function to run after a test.

Reply:
I'll check if there is a way to make this less wonky! My initial thought was to use different target profiles and go down the `TestDrive` path, as you suggested, but it would require a bigger refactor. There's some room to reduce confusion in the dbt test suite in general, so I'll link this to https://github.com/MaterializeInc/materialize-dbt-utils/issues/25 so we remember to pick that back up.

Reply:
Merging with the existing test so we can cut a release with the bugfix.